### PR TITLE
feat: add mutation to update healthcare professional

### LINKED ISSUE #830: Add mutation to update existing healthcare professional
# Problem
We need to have the function to update an existing healthcare professional in the store and we currently do not have it.

# Success
The mutation to update a healthcare professional is added to the `healthcareProfessionalStore.ts`.
### Requirements

- [x] Add mutation to the `healthcareProfessionalStore.ts`

### PR DESCRIPTION
**Pull request by NabbeunNabi**
Resolves #830 
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specs

## 🔧 What changed
This adds a mutation to the store to update Healthcare Professionals.

## 🧪 Testing instructions
Tests will be added in #804 when proper research of mocking `graphQL` request have been researched and implemented

### CODE REVIEW

File: stores/healthcareProfessionalsStore.ts [hunk 1]
```diff
@@ -15,9 +15,22 @@ export const useHealthcareProfessionalsStore = defineStore(
             healthcareProfessionalsData.value = healthcareProfessionalResults
         }
 
+        async function updateHealthcareProfessional(healthcareProfession: MutationUpdateHealthcareProfessionalArgs) {
+            try {
+                return await graphQLClientRequestWithRetry(
+                    gqlClient.request.bind(gqlClient),
+                    updateHealthcareProfessionalGqlMutation,
+                    healthcareProfession
```
Comment (theyokohamalife) [inline stores/healthcareProfessionalsStore.ts:23]:
```suggestion
                    healthcareProfessional
```

File: stores/healthcareProfessionalsStore.ts [hunk 2]
```diff
@@ -34,7 +47,7 @@ async function queryHealthcareProfessionals() {
             (getAllHealthcareProfessionalsData, searchHealthcareProfessionalsData)
         return response?.healthcareProfessionals ?? []
     } catch (error) {
-        console.log(`Error querying the submissions: ${JSON.stringify(error)}`)
+        console.log(`Error querying the healthcare professionals: ${JSON.stringify(error)}`)
```
Comment (theyokohamalife) [inline stores/healthcareProfessionalsStore.ts:50]: Nice! This is much clearer!

File: stores/healthcareProfessionalsStore.ts (hunk 1 above)
Comment (NabbeunNabi) [reply stores/healthcareProfessionalsStore.ts:23]: whoops! Thanks for the catch. Definitely meant to do professional

**Review by theyokohamalife — APPROVED** [review]
LGTM

**Pull request merged**